### PR TITLE
Daan

### PR DESCRIPTION
**Pull request by gielvzwieten**
aaa

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,15 +0,0 @@
-root = true
-
-[*]
-charset = utf-8
-end_of_line = lf
-insert_final_newline = true
-indent_style = space
```
Comment (daanvanzwieten) [inline .editorconfig]: add this file to your .gitignore

File: .gitignore [hunk 2]
```diff
@@ -1,12 +0,0 @@
-/node_modules
-/public/hot
-/public/storage
-/storage/*.key
-/vendor
```
Comment (daanvanzwieten) [inline .gitignore]: Add the .idea directory to your .gitignore

File: app/Category.php [hunk 3]
```diff
@@ -1,17 +0,0 @@
-<?php
-
-namespace App;
-
-use Illuminate\Database\Eloquent\Model;
-
-class Category extends Model
```
Comment (daanvanzwieten) [inline app/Category.php]: Add docheader with properties that this model contains.

```
/**
 * @property int $id
 * @property string $name
 * etc...
 */
```

File: app/Comment.php [hunk 4]
```diff
@@ -1,15 +0,0 @@
-<?php
-
-namespace App;
-
-use Illuminate\Database\Eloquent\Model;
-
-class Comment extends Model
```
Comment (daanvanzwieten) [inline app/Comment.php]: docheader with properties (goes for all models from now on)

File: app/Comment.php [hunk 5]
```diff
@@ -1,15 +0,0 @@
...
-{
-    protected $guarded = [];
-
-    public function post()
-    {
-        return $this->belongsTo(Post::class);
-    }
-}
```
Comment (daanvanzwieten) [inline app/Comment.php]: Should a comment contain a user as well? Who does the comment belong to?

File: app/Http/Controllers/BlogController.php [hunk 6]
```diff
@@ -1,21 +0,0 @@
-<?php
-
-namespace App\Http\Controllers;
-
-use Illuminate\Http\Request;
-use App\Post;
-
-class BlogController extends Controller
-{
-    public function getIndex()
```
Comment (daanvanzwieten) [inline app/Http/Controllers/BlogController.php]: Be consistent with your naming index, show etc. instead of getIndex, getSingle

File: app/Http/Controllers/CategoryController.php [hunk 7]
```diff
@@ -1,111 +0,0 @@
...
-
-    /**
-     * Display a listing of the resource.
-     *
-     * @return \Illuminate\Http\Response
-     */
-    public function index()
-    {
-        $this->authorize('viewAny', Category::class);
-        // display a view of all of our categories
```
Comment (daanvanzwieten) [inline app/Http/Controllers/CategoryController.php]: Only place comments where necessary. Reading the code should make things clear most of the time.

File: app/Http/Controllers/CategoryController.php [hunk 8]
```diff
@@ -1,111 +0,0 @@
...
-
-    /**
-     * Store a newly created resource in storage.
-     *
-     * @param  \Illuminate\Http\Request  $request
-     * @return \Illuminate\Http\Response
-     */
-    public function store(Request $request)
-    {
-        $this->authorize('viewAny', Category::class);
```
Comment (daanvanzwieten) [inline app/Http/Controllers/CategoryController.php]: Change the authorize to use the respective (create(?)) policy method. Goes for every mehtod in this (and other) class(es)

File: app/Http/Controllers/CategoryController.php [hunk 9]
```diff
@@ -1,111 +0,0 @@
...
-        //validate
-        $attributes = $request->validate([
```
Comment (daanvanzwieten) [inline app/Http/Controllers/CategoryController.php]: Does $request->validate() return the validated data or all data?

File: app/Http/Controllers/CategoryController.php [hunk 10]
```diff
@@ -1,111 +0,0 @@
...
-
-        //create and store
-        Category::create($attributes);
-
-        //flash message
-        session()->flash('successmessage', 'Created Category');
-
-        // redirect
-        return redirect('/categories');
-
```
Comment (daanvanzwieten) [inline app/Http/Controllers/CategoryController.php]: remove whitespace

File: app/Http/Controllers/CategoryController.php [hunk 11]
```diff
@@ -1,111 +0,0 @@
...
-    }
-
-    /**
-     * Display the specified resource.
-     *
-     * @param  int  $id
-     * @return \Illuminate\Http\Response
-     */
-    public function show($id)
```
Comment (daanvanzwieten) [inline app/Http/Controllers/CategoryController.php]: remove functions that aren't being used

File: app/Http/Controllers/CommentsController.php [hunk 12]
```diff
@@ -1,99 +0,0 @@
...
-use App\Comment;
-
-class CommentsController extends Controller
-{
-    /**
-     * Display a listing of the resource.
-     *
-     * @return \Illuminate\Http\Response
-     */
-    public function index()
```
Comment (daanvanzwieten) [inline app/Http/Controllers/CommentsController.php]: Remove unused methods

File: app/Http/Controllers/CommentsController.php [hunk 13]
```diff
@@ -1,99 +0,0 @@
...
-     */
-    public function store(Request $request)
-    {
-        //validate data
-        $attributes = $request->validate([
-            'name' => 'required',
-            'email' => 'required',
-            'comment' => 'required',
-        ]);
-        $attributes['post_id'] = $request->post_id;
```
Comment (daanvanzwieten) [inline app/Http/Controllers/CommentsController.php]: post_id should be put inside validate() as well. Plus check if the post actually exists.

File: app/Http/Controllers/PagesController.php [hunk 14]
```diff
@@ -1,55 +0,0 @@
-<?php
-
-namespace App\Http\Controllers;
-
-use Mail;
-use Illuminate\Http\Request;
-use App\Http\Requests;
-use \App\Post;
-class PagesController extends Controller
```
Comment (daanvanzwieten) [inline app/Http/Controllers/PagesController.php]: add whtiespace between `use` and `class`

File: app/Http/Controllers/PagesController.php [hunk 15]
```diff
@@ -1,55 +0,0 @@
...
-    public function postContact(Request $request)
-    {
-        // validate
-        $request->validate([
-            'email' => 'required|email',
-            'subject' => 'required',
-            'message' => 'required',
-        ]);
-
-        $data = [
```
Comment (daanvanzwieten) [inline app/Http/Controllers/PagesController.php]: $data can be retrieved from the validated data above

File: app/Http/Controllers/PagesController.php [hunk 16]
```diff
@@ -1,55 +0,0 @@
...
-            'email' => $request->email,
-            'subject' => $request->subject,
-            'bodyMessage' => $request->message,
-        ];
-
-
```
Comment (daanvanzwieten) [inline app/Http/Controllers/PagesController.php]: remove unnecessary whitespace

File: app/Http/Controllers/PostController.php [hunk 17]
```diff
@@ -1,148 +0,0 @@
...
-    /**
-     * Display a listing of the resource.
-     *
-     * @return \Illuminate\Http\Response
-     */
-    public function index()
-    {
-        $this->authorize('viewAny', Post::class);
-
-        $posts = Post::orderBy('id', 'DESC')->paginate(5);
```
Comment (daanvanzwieten) [inline app/Http/Controllers/PostController.php]: you can also use the `latest()` scope instead of orderBy('id', 'DESC'). It orders by, I believe, the created_at column desc.

File: app/Http/Controllers/PostController.php [hunk 18]
```diff
@@ -1,148 +0,0 @@
...
-     * @return \Illuminate\Http\Response
-     */
-    public function store(Request $request)
-    {
-        $this->authorize('viewAny', Post::class);
-        // validate data
-
-        $attributes = request()->validate([
-            'title'       => 'required|max:255',
-            'body'        => 'required',
```
Comment (daanvanzwieten) [inline app/Http/Controllers/PostController.php]: validate the category_id and tags as well

File: app/Http/Controllers/PostController.php [hunk 19]
```diff
@@ -1,148 +0,0 @@
...
-     * @return \Illuminate\Http\Response
-     */
-    public function update(Request $request, Post $post)
-    {
-        $this->authorize('viewAny', $post);
-
-        $attributes = $request->validate([
-            'title' => 'required|max:255',
-            'body' => 'required',
-            'category_id' => 'sometimes|integer',
```
Comment (daanvanzwieten) [inline app/Http/Controllers/PostController.php]: Check for `exists` as well
Validate tags as well

File: app/Policies/CategoryPolicy.php [hunk 20]
```diff
@@ -1,94 +0,0 @@
...
-    
-    /**
-     * Determine whether the user can view any categories.
-     *
-     * @param  \App\User  $user
-     * @return mixed
-     */
-    public function viewAny(User $user)
-    {
-        return $user->id === 1;
```
Comment (daanvanzwieten) [inline app/Policies/CategoryPolicy.php]: I'd rather check for a user's role for example.

File: app/Policies/CategoryPolicy.php [hunk 21]
```diff
@@ -1,94 +0,0 @@
...
-    /**
-     * Determine whether the user can view the category.
-     *
-     * @param  \App\User  $user
-     * @param  \App\Category  $category
-     * @return mixed
-     */
-    public function view(User $user, Category $category)
-    {
-        //
```
Comment (daanvanzwieten) [inline app/Policies/CategoryPolicy.php]: Make use of the other policy methods as well

File: app/Tag.php [hunk 22]
```diff
@@ -1,15 +0,0 @@
...
-
-namespace App;
-
-use Illuminate\Database\Eloquent\Model;
-
-class Tag extends Model
-{
-    protected $guarded = [];
-
-    public function posts()
```
Comment (daanvanzwieten) [inline app/Tag.php]: You could also look into Polymorphic relations. Not necessary in this setup, but could be nice to have if you plan on having more things you could tag.

File: database/migrations/2019_09_13_094526_create_post_tag_table.php [hunk 23]
```diff
@@ -1,35 +0,0 @@
...
-{
-    /**
-     * Run the migrations.
-     *
-     * @return void
-     */
-    public function up()
-    {
-        Schema::create('post_tag', function (Blueprint $table) {
-            $table->bigIncrements('id');
```
Comment (daanvanzwieten) [inline database/migrations/2019_09_13_094526_create_post_tag_table.php]: incrementing id not necessary for this table. 

Make post_id + tag_id a unique index

File: routes/web.php [hunk 24]
```diff
@@ -1,37 +0,0 @@
...
-| Here is where you can register web routes for your application. These
-| routes are loaded by the RouteServiceProvider within a group which
-| contains the "web" middleware group. Now create something great!
-|
-*/
-
-Route::get('/blog/{post}/{slug}', 'BlogController@getSingle');
-Route::get('/blog', 'BlogController@getIndex');
-
-Route::get('/', 'PagesController@getIndex');
```
Comment (daanvanzwieten) [inline routes/web.php]: Be consistent with your naming

File: routes/web.php [hunk 25]
```diff
@@ -1,37 +0,0 @@
...
-| Web Routes
-|--------------------------------------------------------------------------
-|
-| Here is where you can register web routes for your application. These
-| routes are loaded by the RouteServiceProvider within a group which
-| contains the "web" middleware group. Now create something great!
-|
-*/
-
-Route::get('/blog/{post}/{slug}', 'BlogController@getSingle');
```
Comment (daanvanzwieten) [inline routes/web.php]: a {post} and a {slug}? I think it should be either a post id or a slug.

Comment (gielvzwieten) [reply routes/web.php]: slug heb ik toegevoegd als titel van de post in de url, voor seo friendlyness? en omdat 2 titels hetzelfde kunnen zijn heb ik de postid eerst
